### ISSUE TITLE
Remove direct mysql dependency?

### ISSUE THREAD

**Issue by bebraw** [overlookmotel/sequelize-extra#2]
I noticed your project depends on `mysql` directly. Is that absolutely needed?

**Comment by overlookmotel** [comment]
Well, no it doesn't require mysql strictly, if the user is using e.g. Postgres.

But if the DB modules aren't listed as dependencies, the user would have to install them additionally in their app.

I'm not tied to this approach. What would you suggest?

**Comment by overlookmotel** [comment]
In any case, primarily I published this module for my own use. All it really does is package together the three Sequelize plugins I've written (sequelize-definer, sequelize-hierarchy and sequelize-virtual-fields).

I'd imagine for most cases, people would be better off just using the plugins that they actually need.

**Comment by bebraw** [comment]
> I'd imagine for most cases, people would be better off just using the plugins that they actually need.

Yeah, that was the point.

I don't see any direct dependency on `mysql` at `/lib` so I think it might be safe to remove it.

**Comment by overlookmotel** [comment]
No, sorry I meant, people would be better off just using sequelize-definer, sequelize-hierarchy and sequelize-virtual-fields directly. None of those have the db library dependencies.

**Comment by bebraw** [comment]
> No, sorry I meant: people would be better off just using sequelize-definer, sequelize-hierarchy and sequelize-virtual-fields directly. None of those have the db library dependencies.

Ok, you are free to do however you want. I would push `mysql` to dev-deps given you are using it at tests only.

One alternative would be to push `Instance#getValues()` and such to another module that can be used standalone. After that this module would just become a proxy for your utility modules. This would allow you to use `sequelize-extra` like currently while making it possible to pick some specific bit of the whole.

**Comment by overlookmotel** [comment]
Yes, the latter is a really good idea. I'll do that when I get the chance.

Also waiting on decisions from sequelize core contribs on how they want to standardize how plugins integrate with Sequelize, so once that's decided, I'll do a bit of a spring clean and get all these modules a bit more ship-shape.

**Comment by overlookmotel** [comment]
Out of interest, are you wanting to use this module for the `Instance#getValues()` etc methods? Or for some of the functionality from sequelize-definer, sequelize-hierarchy and sequelize-virtual-fields?

**Comment by bebraw** [comment]
I'm doing little REST APIs on top of sequelize (example, [swagger-todo](https://github.com/bebraw/swagger-todo)) and it seemed to me that your work has potential to simplify some of mine. For now `Sequelize.getValues`, sequelize-definer (simplifies definitions) and sequelize-virtual-fields (derived fields!) should help a lot!

**Comment by overlookmotel** [comment]
Hi @bebraw.

I finally got round to spitting off the `getValues` functionality into it's own module [sequelize-values](https://www.npmjs.org/package/sequelize-values).

So now there should be no need for anyone to utilize this sequelize-extra module. I've added a note in the README to this effect.

If you just use [sequelize-values](https://www.npmjs.org/package/sequelize-values) yourself, you won't need to include the mysql dependency, so that'll solve this problem.

I'm closing this issue as I think this resolves the problem.

Thanks!
